File: golden_repo/mdforge/config/defaults.py
"""Default configuration values for mdforge."""

DEFAULTS = {
    # Maximum file size to process (in bytes)
    "max_file_size": 10 * 1024 * 1024,  # 10MB

    # Supported file extensions
    "markdown_extensions": [".md", ".markdown", ".mdown", ".mkd"],

    # Default output format for stats
    "stats_format": "json",

    # Timeout for batch operations (seconds)
    "batch_timeout": 30,

    # Maximum recursion depth for directory scanning
    "max_depth": 10,

    # Whether to follow symlinks
    "follow_symlinks": False,

    # Default encoding
    "encoding": "utf-8",

    # HTML conversion options
    "html": {
        "wrap_in_body": False,
        "include_doctype": False,
        "escape_html": True,
    },

    # TOC generation options
    "toc": {
        "min_level": 2,
        "max_level": 4,
        "ordered": False,
    },
}


def merge_config(user_config: dict | None = None) -> dict:
    """Merge user config with defaults (shallow)."""
    if user_config is None:
        user_config = {}
    return {**DEFAULTS, **user_config}

File: source_repo/src/config/defaults.js
'use strict';

/**
 * Default configuration values for mdforge.
 */
const DEFAULTS = {
  // Maximum file size to process (in bytes)
  maxFileSize: 10 * 1024 * 1024, // 10MB

  // Supported file extensions
  markdownExtensions: ['.md', '.markdown', '.mdown', '.mkd'],

  // Default output format for stats
  statsFormat: 'json',

  // Timeout for batch operations (ms)
  batchTimeout: 30000,

  // Maximum recursion depth for directory scanning
  maxDepth: 10,

  // Whether to follow symlinks
  followSymlinks: false,

  // Default encoding
  encoding: 'utf-8',

  // HTML conversion options
  html: {
    wrapInBody: false,
    includeDoctype: false,
    escapeHtml: true,
  },

  // TOC generation options
  toc: {
    minLevel: 2,
    maxLevel: 4,
    ordered: false,
  },
};

/**
 * Merge user config with defaults (shallow).
 * @param {object} userConfig
 * @returns {object}
 */
function mergeConfig(userConfig = {}) {
  return { ...DEFAULTS, ...userConfig };
}

module.exports = { DEFAULTS, mergeConfig };